File: src/DeployManager.Domain/Entities/DeploySite.cs
using DeployManager.Domain.Common;
using DeployManager.Domain.Enums;

namespace DeployManager.Domain.Entities;

public class DeploySite : BaseEntity
{
    public string Code { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public Guid EnvironmentId { get; set; }
    public DeployTargetType TargetType { get; set; }
    public string RootPath { get; set; } = string.Empty;
    public string? PublicUrl { get; set; }
    public string? Username { get; set; }
    public string? PasswordEncrypted { get; set; }
    public bool IsActive { get; set; } = true;
    public DeployEnvironment? Environment { get; set; }
}

File: src/DeployManager.Client/src/shared/types/deploySite.ts
export interface DeploySite {
  id: string
  code: string
  name: string
  environmentId: string
  environmentName: string
  targetType: string
  rootPath: string
  publicUrl: string | null
  username: string | null
  isActive: boolean
  createdAt: string
  updatedAt: string | null
}

export interface CreateDeploySiteRequest {
  code: string
  name: string
  environmentId: string
  targetType: string
  rootPath: string
  publicUrl?: string
  username?: string
  password?: string
}

export interface UpdateDeploySiteRequest extends CreateDeploySiteRequest {
  id: string
  isActive: boolean
}
